ts/components/ui/ListItemInfo.tsx

```diff
@@ -0,0 +1,137 @@
+import { Body } from "../core/typography/Body";
+import { VSpacer } from "../core/spacer/Spacer";
+
+export type ListItemInfo = WithTestID<{
+  label: string;
+  value: string | React.ReactNode;
+  icon?: IOIcons;
+  // Accepted components: ButtonLink, IconButton
+  // Don't use any components other than these
+  action?: React.ReactNode;
+  // Accessibility
+  accessibilityLabel: string;
+}>;
+
+const styles = StyleSheet.create({
+  textValue: {
+    fontSize: 18,
+    lineHeight: 24,
+    ...makeFontStyleObject("SemiBold", undefined, "TitilliumWeb")
+  }
+});
+
+export const ListItemInfo = ({
+  label,
+  value,
+  icon,
+  action,
+  accessibilityLabel,
+  testID
+}: ListItemInfo) => {
+  const isDesignSystemEnabled = useIOSelector(isDesignSystemEnabledSelector);
+
+  const theme = useIOTheme();
+
+  /* ◀ REMOVE_LEGACY_COMPONENT: Start */
+  const LegacyListItemNav = () => (
+    <View
+      style={IOListItemStyles.listItem}
+      testID={testID}
+      accessible={true}
+      accessibilityLabel={accessibilityLabel}
+    >
+      <View style={IOListItemStyles.listItemInner}>
+        {icon && (
+          <View style={{ marginRight: IOListItemVisualParams.iconMargin }}>
+            <Icon
+              name={icon}
+              color="grey-450"
+              size={IOListItemVisualParams.iconSize}
+            />
+          </View>
+        )}
+        <View style={IOStyles.flex}>
+          {/* Let developer using a custom component (e.g: skeleton) */}
+          {typeof label === "string" ? (
+            <Body weight="Regular">{label}</Body>
+          ) : (
+            { label }
+          )}
+          {typeof value === "string" ? (
+            <Text
+              style={[styles.textValue, { color: IOColors.bluegreyDark }]}
+              numberOfLines={2}
+            >
+              {value}
+            </Text>
+          ) : (
+            { value }
+          )}
+        </View>
+        {action && (
+          <View style={{ marginLeft: IOListItemVisualParams.iconMargin }}>
+            {action}
+          </View>
+        )}
+      </View>
+    </View>
+  );
+  /* REMOVE_LEGACY_COMPONENT: End ▶ */
+
+  const NewListItemNav = () => (
+    <View
+      style={IOListItemStyles.listItem}
+      testID={testID}
+      accessible={true}
+      accessibilityLabel={accessibilityLabel}
+    >
+      <View style={IOListItemStyles.listItemInner}>
+        {icon && (
+          <View style={{ marginRight: IOListItemVisualParams.iconMargin }}>
+            <Icon
+              name={icon}
+              color="grey-450"
+              size={IOListItemVisualParams.iconSize}
+            />
+          </View>
+        )}
+        <View style={IOStyles.flex}>
+          <LabelSmall weight="Regular" color={theme["textBody-tertiary"]}>
+            {label}
+          </LabelSmall>
+          <VSpacer size={4} />
+          <NewH6 color={theme["textBody-default"]} numberOfLines={2}>
+            {value}
+          </NewH6>
```

Comment:
What do you think if the `numberOfLines` is customizable by the component prop that have `2` as default value? It can be useful sometimes or in the future to be able also to choose how many number of lines we want if different from 2

Reply:
I just added it with the default value set to `2`. Note that the `value` prop also accepts `ReactNode`, so you can always add a specific component for the edge cases. In this context, you reminded me of the case with the physical address (and relative newlines), so I've added a specific example with newlines managed through `\n` (as a string).